### ISSUE TITLE
503 error (timeout) on Heroku

### ISSUE THREAD

**Issue by szeryf** [eliotsykes/rack-zippy#37]
I added rack-zippy to my app hosted on Heroku (Ruby 2.1, Rails 4.0.5), but can't get it to work. When I refresh the page, my `application.js` file is not served, instead I get a 503 error meaning the request took too long (more than 30s) and heroku killed it:

```
2014-09-02T12:36:17.100262+00:00 heroku[router]: at=error code=H12 desc="Request timeout"
method=GET path="/assets/application-9672ef404f5cafb00b5883491071a7aa.js" 
host=tenbilliondreams-staging.herokuapp.com request_id=d885753e-b7e2-4aa4-b28d-f01473939ad6 
fwd="37.128.4.170" dyno=web.1 connect=0ms service=30001ms status=503 bytes=1123
```

The file was served with no problems before I added rack-zippy and when I remove or disable it, it's served again. Other files (e.g. `application.css`) work fine with rack-zippy. 

I thought this could be a problem with the file size, i.e. my `application.js` is so big (it's currently 2.9 MB) it could not be compressed in under 30s? But if I understand correctly, rack-zippy is supposed to serve already precompiled compressed files, not compress the on the fly (and I checked, both `application-<hash>.js.gz` and `application-<hash>.css.gz` are present in `public/assets` on Heroku). So, any other ideas?

**Comment by bismark64** [comment]
Any update for this issue? Thanks

**Comment by eliotsykes** [comment]
Hi @bismark64 - no updates I'm afraid. If you're able to help with contributing a pull request or further details on how to reproduce this issue reliably that'd be sincerely appreciated.

**Comment by eliotsykes** [comment]
Closing as may be fixed in upcoming 4.0 rack-zippy release. Reopen or log new issue if you see this on 4.0
